src/ansys/geometry/core/misc/auxiliary.py

```diff
@@ -237,6 +238,28 @@ def get_edges_from_ids(design: "Design", edge_ids: list[str]) -> list["Edge"]:
     ]  # noqa: E501
 
 
+def get_beams_from_ids(design: "Design", beam_ids: list[str]) -> list["Beam"]:
+    """Find the ``Beam`` objects inside a ``Design`` from its ids.
+
+    Parameters
+    ----------
+    design : Design
+        Parent design for the beams.
+    beam_ids : list[str]
+        List of beam ids.
+
+    Returns
+    -------
+    list[Beam]
+        List of Beam objects.
+
+    Notes
+    -----
+    This method takes a design and beam ids, and gets their corresponding ``Beam`` objects.
+    """
+    return [beam for beam in design.beams if beam.id in beam_ids]  # noqa: E501
+
```

Comment:
Hmm... we might need to traverse all components inside the design.. beams belong to components. They won't be only available at design level.